### PR TITLE
[SettingsBundle] Add post and pre save events on settings persistance

### PR DESCRIPTION
**Pull request by aramalipoor**
Q  | A
------------- | -------------
Bug fix?  | no
New feature?  | yes
BC breaks? | no
Deprecations? | no
Fixed tickets | -
License | MIT
Doc PR | -

### CODE REVIEW

**Comment by pjedrzejewski** [comment]
:+1:

**Comment by aRn0D** [comment]
@pjedrzejewski why is there not a settings component ?

File: src/Sylius/Bundle/SettingsBundle/Event/SettingsEvent.php [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+use Symfony\Component\EventDispatcher\GenericEvent;
+
+/**
+ * Settings event.
+ *
+ * @author Aram Alipoor <aram.alipoor@gmail.com>
+ */
+class SettingsEvent extends GenericEvent
+{
+    const PRE_SAVE = 'sylius.settings.pre_save';
```
Comment (aRn0D) [inline src/Sylius/Bundle/SettingsBundle/Event/SettingsEvent.php]: This should be moved in a class like `SyliusOrderEvents`for example.

**Comment by aRn0D** [comment]
:+1:

**Comment by pjedrzejewski** [comment]
@aramalipoor Thanks! @aRn0D Cause this component was quite striclty connected to Symfony and Doctrine, that being said I think we should create one now. :)

**Pull request merged**